jax/experimental/export/export.py

```diff
@@ -1072,12 +1104,47 @@ def convert_shape(x: ir.Value, x_aval: core.AbstractValue, new_aval: core.Abstra
                              kept_args)
   # The ctx.avals_out already contain the abstract values refined by
   # _call_exported_abstract_eval.
-  return tuple(
+  results = tuple(
       convert_shape(out, out_aval, refined_out_aval)
       for out, out_aval, refined_out_aval in zip(call.results, exported.out_avals, ctx.avals_out))
+  # Apply out_shardings
+  results = tuple(
+    wrap_with_sharding(ctx, exported, x, x_aval, x_sharding)
+    for x, x_aval, x_sharding in zip(results, ctx.avals_out, exported.out_shardings)
+  )
+  return results
 
 
 for _p in ("cpu", "tpu", "cuda", "rocm"):
   mlir.register_lowering(call_exported_p,
                          functools.partial(_call_exported_lowering, platform=_p),
                          platform=_p)
+
+def wrap_with_sharding(ctx: mlir.LoweringRuleContext,
+                       exported: Exported,
+                       x: ir.Value,
+                       x_aval: core.AbstractValue,
+                       x_sharding: Sharding) -> ir.Value:
+  if sharding_impls.is_unspecified(x_sharding):
+    return x
+  if not isinstance(x_sharding, sharding_impls.GSPMDSharding):
```

Comment:
With the changes I mentioned below, you can remove this check too.

No need to restrict on just GSPMDSharding but I agree that lowering will only see GSPMDSharding. Still no need for this check.

Reply:
Done